### ISSUE TITLE
Add support for Android >= 33

### ISSUE THREAD

**Issue by tmtit** [amorenew/Flutter-Mobile-Number-Plugin#48]
I used your library to get the phone number, but it doesn't work with eSIM. I tried replacing 
`subscriptionInfo.getNumber();`
with
`final int id = SubscriptionManager.getActiveDataSubscriptionId();
final String phoneNumber = subscriptionManager.getPhoneNumber(id);`
and it worked. This method was added in API 33 to replace the deprecated getNumber()

doc: https://developer.android.com/reference/kotlin/android/telephony/SubscriptionManager#getphonenumber_1

**Comment by amorenew** [comment]
@tmtit How could I test it on an emulator, I don’t have esim on my personal phone?

**Comment by bsutton** [comment]
When trying to build with gradle 8.1, api version 35. I get the following error:

```
global options.
bsutton@slayer4:~/git/hmb$ flutter build apk


FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':mobile_number'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

     If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 59s
Running Gradle task 'assembleRelease'...                          120.0s
Gradle task assembleRelease failed with exit code 1

```

**Comment by bsutton** [comment]
google suggests: 

 Add the namespace property to the android block of the module's build.gradle file. The value should match the package attribute in the AndroidManifest.xml.

android {
    namespace 'com.yourpackage.mobile_number'
    ...
}

**Comment by amorenew** [comment]
@bsutton fixed on release 3.0.0
https://pub.dev/packages/mobile_number

**Comment by bsutton** [comment]
Thanks
